dev/ci/scripts/utils/ci_utils.sh

```diff
@@ -122,10 +122,19 @@ function cancel_all_batch_jobs () {
 function create_experiment () {
 
   local yaml_config="${1}"
+  local TAG="${2}"
   cd "${HOMEgfs_}" || exit 1
   pr_sha=$(git rev-parse --short HEAD)
   case=$(basename "${yaml_config}" .yaml) || true
-  export pslot=${case}_${pr_sha}
+  
+  # Use TAG if provided as second argument, otherwise use pr_sha
+  if [[ -n "${TAG}" ]]; then
```

Comment:
If `TAG` is optional, then a default should be given:

```suggestion
  local TAG="${2:}"
  cd "${HOMEgfs_}" || exit 1
  pr_sha=$(git rev-parse --short HEAD)
  case=$(basename "${yaml_config}" .yaml) || true

  # Use TAG if provided as second argument, otherwise use pr_sha
  if [[ -n "${TAG}" ]]; then

Reply:
The option is not to use it so there is _no_ default.  The logic is to then use a hash.

Reply:
The `-n` option shouldn't be used on an undefined variable.  Similarly, `TAG` shouldn't be defined to `$2` if a second option is not passed.  Both of these will fail if `set -u` is applied and I think we should try to eliminate referencing unassigned variables.

If you pass an empty string to `TAG`, then it will be assigned an empty string.  This will trigger the `else` in this block and avoid using unassigned variables.

Reply:
Ok, let me fix this up.  Didn't know about this level of detail regarding BASH arguments and behavior.  Very useful information for stable BASH code for sure.

Reply:
Oh I see, you added the : ok now I get it.